src/nft/components/details/detailsV2/BuyButton.tsx

```diff
@@ -24,30 +87,136 @@ const Price = styled.div`
   color: ${({ theme }) => theme.accentTextLightSecondary};
 `
 
-export const BuyButton = ({ asset, onDataPage }: { asset: GenieAsset; onDataPage?: boolean }) => {
+export const BuyButton = ({ asset, dataPage: dataPage }: { asset: GenieAsset; dataPage?: boolean }) => {
+  const { account } = useWeb3React()
+  const [accountDrawerOpen, toggleWalletDrawer] = useAccountDrawer()
   const { fetchAndPurchaseSingleAsset, isLoading: isLoadingRoute } = useBuyAssetCallback()
+  const [connectingToWallet, setConnectingToWallet] = useState(false)
+  const [addToBagExpanded, setAddToBagExpanded] = useState(false)
+  const navigate = useNavigate()
+  const setSellPageState = useProfilePageState((state) => state.setProfilePageState)
+  const resetSellAssets = useSellAsset((state) => state.reset)
+  const clearCollectionFilters = useWalletCollections((state) => state.clearCollectionFilters)
+
+  const { walletAssets } = useNftBalance(account ?? '', [], [{ address: asset.address, tokenId: asset.tokenId }], 1)
+  const walletAsset = walletAssets?.[0]
+
   const price = asset.sellorders?.[0]?.price.value
+  const ownsAsset = Boolean(walletAsset) && account?.toLowerCase() === asset.ownerAddress?.toLowerCase()
+
+  const assetInBag = useIsAssetInBag(asset)
+  const { addAssetsToBag, removeAssetsFromBag } = useBag(
+    ({ addAssetsToBag, removeAssetsFromBag }) => ({
+      addAssetsToBag,
+      removeAssetsFromBag,
+    }),
+    shallow
+  )
+
+  const navigateToProfile = useCallback(() => {
+    resetSellAssets()
+    setSellPageState(ProfilePageStateType.VIEWING)
+    clearCollectionFilters()
+    navigate('/nfts/profile')
+  }, [clearCollectionFilters, navigate, resetSellAssets, setSellPageState])
+
+  const oneClickBuyAsset = useCallback(() => {
+    if (!account) {
+      if (!accountDrawerOpen) {
+        toggleWalletDrawer()
+      }
+      setConnectingToWallet(true)
+      setTimeout(() => setConnectingToWallet(false), 20000)
```

Comment:
What do you think about darkening the side button as well while waiting?
![Screen Shot 2023-05-31 at 16 09 07 ](https://github.com/Uniswap/interface/assets/11512321/44a51126-3540-4a5d-ad76-5cf68421fb50)

Reply:
I actually asked phil about this and he liked this way